lib/build.py

```diff
@@ -221,6 +221,7 @@ def __default__(self, tree):
                ch.FATAL("first instruction must be ARG or FROM")
          inst.execute()
          if (image_i != -1):
+            inst.metadata_history_update(self.instruction_ct)
```

Comment:
This would be better in `execute()` I think. Then it wouldn't need to care about the instruction index.